lib/matplotlib/dates.py

```diff
@@ -1241,10 +1241,7 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
-        if interval < 1:
-            raise ValueError("The interval parameter must be an integer "
-                             "greater than or equal to one.")
-        if not isinstance(interval, int):
+        if interval < 1 or not isinstance(interval, int):
             raise ValueError("The interval parameter must be an integer "
                              "greater than or equal to one.")
```

Comment:
```python
$ python2 
>>> 1 < '0'
True

$ python3
>>> 1 < '0'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unorderable types: int() < str()
```
Python 2 will compare incompatible types by the name of the type (which is completely arbitrary); Python 3 is not so forgiving and throws an error.